### PR TITLE
nextcloud: restrict web server support to nginx; stop sharing nginx user/group; improve setup service

### LINKED ISSUE #48881: nextcloud doesn't create the needed groups
## Issue description

I tried the new nextcloud service on nixos-unstable but seems like it lacks an assert or forgets to create group for nginx since this simple config

```
{ config, lib, pkgs,  ... }:
let
  secrets = import ../secrets.nix;
in
{
  services.nextcloud = {
    enable = true;
    # TODO update later
    hostName = "toto.com";
    # nginx.enable = true;
    config = {
      adminpass = secrets.nextcloud.password;
    };
    maxUploadSize = "512M";
  };
}

```
generates a deployment failure (deployed via nixops):
```
mptcp-server> Oct 23 22:10:40 mptcp systemd[1]: Starting PHP FastCGI Process Manager service for pool nextcloud...
mptcp-server> Oct 23 22:10:40 mptcp php-fpm[8850]: [ERROR] [pool nextcloud] cannot get gid for group 'nginx'
mptcp-server> Oct 23 22:10:40 mptcp php-fpm[8850]: [ERROR] [pool nextcloud] cannot get gid for group 'nginx'
mptcp-server> Oct 23 22:10:40 mptcp php-fpm[8850]: [ERROR] FPM initialization failed
mptcp-server> Oct 23 22:10:40 mptcp php-fpm[8850]: [ERROR] FPM initialization failed
```

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.73, NixOS, 19.03.git.50dfcd2 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(teto): `""`
 - channels(root): `"nixos-unstable-18.03pre130932.cc4677c36ee"`
 - nixpkgs: `/home/teto/nixpkgs`
```

### LINKED ISSUE #69261: nextcloud: php-fpm and nextcloud-setup services fail when nginx is not enabled
**Describe the bug**
When deploying a nextcloud service, if nginx is not enabled, the unit scripts of php-fpm and nextcloud-setup systemd services fail. This makes it more difficult than necessary to use nextcloud with a different web server.

For both services, the issue is caused by hardcoding the nginx user and/or group in the unit scripts.

For the php-fpm service, we see nginx hardcoded in a [few](https://github.com/NixOS/nixpkgs/blob/3c10b93320b69fdb3df24e3f0fbcbe939bf0ead3/nixos/modules/services/web-apps/nextcloud.nix#L429) [places](https://github.com/NixOS/nixpkgs/blob/3c10b93320b69fdb3df24e3f0fbcbe939bf0ead3/nixos/modules/services/web-apps/nextcloud.nix#L437).

For nextcloud-setup, the failure is related to [this line](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/web-apps/nextcloud.nix#L399), where the nginx group is hardcoded into this chown command.

The resulting error message:
```
tshaynik> ● phpfpm-nextcloud.service - PHP FastCGI Process Manager service for pool nextcloud
tshaynik>    Loaded: loaded (/nix/store/2zbp405ils6dbzcclqyr3zf2zrhrk3ch-unit-phpfpm-nextcloud.service/phpfpm-nextcloud.service; enabled; vendor preset: enabled)
tshaynik>    Active: failed (Result: exit-code) since Sun 2019-09-22 17:50:06 UTC; 26s ago
tshaynik>   Process: 4977 ExecStart=/nix/store/r9r5cr97ih4qvlnsh8nzk8id0ffxy6r7-php-7.3.9/bin/php-fpm -y /nix/store/v0rwhdgl0xd073frzl30kjbgkr8dmhyk-phpfpm-nextcloud.conf -c /nix/store/j5w50abdyrafgdng9j60jardifa0nbpi-php.ini (code=exited, status=78)
tshaynik>   Process: 4975 ExecStartPre=/nix/store/r2xhb14kz57hmycvjsvwgpzkl8kd3l2g-unit-script-phpfpm-nextcloud-pre-start (code=exited, status=0/SUCCESS)
tshaynik>  Main PID: 4977 (code=exited, status=78)
tshaynik>
tshaynik> Sep 22 17:50:05 tshaynik systemd[1]: Starting PHP FastCGI Process Manager service for pool nextcloud...
tshaynik> Sep 22 17:50:06 tshaynik php-fpm[4977]: [ERROR] [pool nextcloud] cannot get gid for group 'nginx'
tshaynik> Sep 22 17:50:06 tshaynik php-fpm[4977]: [ERROR] [pool nextcloud] cannot get gid for group 'nginx'
tshaynik> Sep 22 17:50:06 tshaynik php-fpm[4977]: [ERROR] FPM initialization failed
tshaynik> Sep 22 17:50:06 tshaynik php-fpm[4977]: [ERROR] FPM initialization failed
tshaynik> Sep 22 17:50:06 tshaynik systemd[1]: phpfpm-nextcloud.service: Main process exited, code=exited, status=78/CONFIG
tshaynik> Sep 22 17:50:06 tshaynik systemd[1]: phpfpm-nextcloud.service: Failed with result 'exit-code'.
tshaynik> Sep 22 17:50:06 tshaynik systemd[1]: Failed to start PHP FastCGI Process Manager service for pool nextcloud.
tshaynik>
tshaynik> ● nextcloud-setup.service
tshaynik>    Loaded: loaded (/nix/store/4asb31g7w00jrr1kc0k60xi4nigh9bga-unit-nextcloud-setup.service/nextcloud-setup.service; enabled; vendor preset: enabled)
tshaynik>    Active: failed (Result: exit-code) since Sun 2019-09-22 17:50:05 UTC; 26s ago
tshaynik>   Process: 4961 ExecStart=/nix/store/yc3bb7dxnqpkca3h1zkf8pb5fbhbak1r-unit-script-nextcloud-setup-start (code=exited, status=1/FAILURE)
tshaynik>  Main PID: 4961 (code=exited, status=1/FAILURE)
tshaynik>
tshaynik> Sep 22 17:50:05 tshaynik systemd[1]: Starting nextcloud-setup.service...
tshaynik> Sep 22 17:50:05 tshaynik yc3bb7dxnqpkca3h1zkf8pb5fbhbak1r-unit-script-nextcloud-setup-start[4961]: chown: invalid group: ‘nextcloud:nginx’
```

**To Reproduce**
1. Enable the `nextcloud` service, and add the `nextcloud-setup` systemd service as described in the [NixOS manual](https://nixos.org/nixos/manual/index.html#module-services-nextcloud), but with the`nginx.enable` option set to false.

``` Nix
    services.nextcloud = {
      enable = true;
      hostName = "nextcloud.example.com";
      nginx.enable = false;
      config = {
        dbtype = "pgsql";
        dbuser = "nextcloud";
        dbhost = "/tmp"; # nextcloud will add /.s.PGSQL.5432 by itself
        dbname = "nextcloud";
        adminpassFile = "/run/keys/nextcloud";
        adminuser = "root";
      };
    };

    systemd.services."nextcloud-setup" = {
      requires = ["postgresql.service"];
      after = ["postgresql.service"];
    };

...
```

**Expected behavior**

Not enabling nginx should still allow nextcloud to be deployed, and the user should be able to override the webserver group. It seems to me like there should be an additional option for the name of the webserver's user and group, that would default to nginx, but could changed by the user to something else.

**Additional context**
Running using nixops on DigitalOcean.

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.
``` yaml
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.72, NixOS, 19.03.173506.2dfae8e22fd (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(mandolin): `"unstable-19.09pre185176.1036dc66416"`
 - channels(root): `"nixos-19.03.173506.2dfae8e22fd"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by DavHau**
###### Things done
(This description is heavily outdated. Plans changed a lot due to the discussions. See below)
 - user and group used by all nextcloud related services are now unified and configurable. If nginx is enabled, the group will be forced to `nginx` and an error is raised if group is set manually.
- nginx is enabled by default now. (In favor of having a working default configuration)
- phpfpm also uses configured nextcloud user + group
- In the setup service instead of `mkdir` + `chown` now `install` is used to setup the directories.
  This has the disadvantage that the group of nextcloud cannot be changed anymore on an existing installation without having to fix the ownership of files manually. But it think `chown`ing all files on every single startup is bad since it's a very intense operation and can take very long depending on the amount of files stored. Also the `chown` operation itself might not be allowed depending on the filesystem. I for example use a sshfs mount for my nextcloud files where this fails

fixes #69261
fixes #48881

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [inline nixos/modules/services/web-apps/nextcloud.nix:76]: What is the purpose of a configurable `user` option?

If the user is configurable then we should follow the pattern of conditionally provisioning users only if the default value is specified:
```
users.users.nextcloud = optionalAttrs (cfg.user == "nextcloud") {
  ...
};
```

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [inline nixos/modules/services/web-apps/nextcloud.nix:529]: This isn't what we want. The `user` option should specify the user that the `nextcloud` service runs as, not the user the web server runs as. `"listen.owner"` is intended to be the user the web server is run as.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [inline nixos/modules/services/web-apps/nextcloud.nix:489]: This script really shouldn't even be run as `root`, so if there are any changes here maybe it should be to fix that...

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [inline nixos/modules/services/web-apps/nextcloud.nix:81]: Another possibility here would be to keep running `nextcloud` services (including `php-fpm`) as a non web server user (like the current `nextcloud` user) and instead add the `nextcloud` group to your web servers `extraGroups` option, like so:
```
mkIf cfg.nginx.enable {
  users.users.nginx.extraGroups = [ cfg.group ]; # in other words "nextcloud"
}
```

The reason for this is that the `nginx` user (and group) likely has access to very sensitive files, such as certificates. If `nginx` can read `nextcloud` files (which you want) that is much less of a risk than having `nextcloud` able to read `nginx` files - a serious security concern. This obviously requires that your `nextcloud` files are `g+r`.

**Review by aanderse — COMMENTED** [review]
Here are some initial thoughts I had which might be a starting point for further discussion.

**Comment by Ma27** [comment]
Did anyone of you test what happens when deploying this configuration to an existing Nextcloud instance? Also, what's supposed to happen if `cfg.user` changes (or someone switches e.g. from `nginx` to `httpd`?).

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:81]: Sounds like a good idea. The files created by nextcloud seem to be `g+r`, so this shouldn't be an issue.

Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:81]: I wonder how to best change the user/group of nextcloud without breaking existing installations. It would require a one time service run as root which changes the ownership. And I would need to have some state file to save that this setup has been completed.
Alternatively I could leave the default user.group to be nginx if `system.stateVersion <= 20.03`, but i don't really like that approach.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:76]: Actually the user doesn't really need to be configurable. Sharing the group with other services which need to access the same files should be sufficient. So I will revert this change.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:529]: Then I should change the user back to `nextcloud` (due to the other discussions).
And the `listen.owner/group` should be set to `config.services.nginx.user/group` if nginx is enabled.
But what if nginx is disabled? Should we add optional options to configure the owner/group for the webservice?

**Comment by DavHau** [comment]
please check https://github.com/NixOS/nixpkgs/pull/93584/commits/6ee30041327343137bab79cf924eac1a17ad0d5c.
  - removed optons cfg.user, cfg.groups
  - add option `serverUser` which is required when not using nginx
  - add `serverUser` to nextcloud group
  - set user/group to "nextcloud" for nextcloud services
  - make setup-service non-root

Did not yet implement a solution for fixing group ownership of files for existing installations. Your suggestions are welcome.

**Comment by aanderse** [comment]
@DavHau for context how are you trying to run `nginx`? `caddy`?

**Comment by DavHau** [comment]
Sorry my description of what I changed was not very precise.

All nextcloud services now use the user and group "nextcloud" (hard-coded). The web server user must therefore be part of the the nextcloud group.

The webserver user can be configured via `cfg.serverUser`.
This user will be automatically added to the nextcloud group and it will be used for `listen.owner` of phpfpm. Also its group will automatically be used for `listen.group`

`cfg.serverUser`defaults to the nginx user if nginx is enabled. Otherwise it must be specified manually.

An example configuration using nginx would be:  
(if nothing specified nginx will be enabled and its user added to the nextcloud group)
```nix
{ pkgs, ... }:
{
  networking.firewall.allowedTCPPorts = [80];
  services.nextcloud = {
    enable = true;
    hostName = "10.233.1.2";
    config.adminpass = "test";
    package = pkgs.nextcloud19;
  };
}
```

A configuration for an alternative webserver could  look like this:
```nix
{ pkgs, ... }:
{
  networking.firewall.allowedTCPPorts = [80];
  services.nextcloud = {
    enable = true;
    hostName = "10.233.1.2";
    config.adminpass = "test";
    package = pkgs.nextcloud19;
    
    # disable nginx and set serverUser
    nginx.enable = false;
    serverUser = "caddy";

    # caddy config
    ...
  };
}
```

**Comment by aanderse** [comment]
@DavHau by having a configurable web server user in this module it seems we're just circling back around to https://github.com/NixOS/nixpkgs/issues/93070.

@Ma27 I suggest we remove the `nginx.enable` option entirely and just take it as a given: this module is for `nginx` only.

**Comment by DavHau** [comment]
- removed option `nextcloud.nginx.enable`. Nginx is now the only supported web server.
- modified nextcloud-setup service to fix group ownership of nextcloud files if they don't match.
  All installations prior to this PR will be fixed by this. (changed from nginx -> nextcloud).

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix]: Forget about it. Since it looks like the user will still be `nextcloud`, only the group needs to be changed from `nginx` to `nextcloud` which is easy to do in the current setup. Also no state needs to be stored.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [inline nixos/modules/services/web-apps/nextcloud.nix:513]:
```suggestion
            "listen.group" = config.services.nginx.group;
```

**Review by aanderse — CHANGES_REQUESTED** [review]
Looking good :+1:

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [inline nixos/modules/services/web-apps/nextcloud.nix:468]: This script is run as `nextcloud` so this `chown` doesn't make sense to me. The code this script replaces also had that, though... can someone explain this to me?

Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:468]: The chown command is only for fixing the group permissions for installation prior to this PR.
Those installations will have `nextcloud:nginx` and it will be changed to `nextcloud:nextcloud`. This command is possible as nextcloud user.

Other reasons why I wrote the script like this:
1. We can't just do a plain `install -o nextcloud -g nextcloud -d ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps`. Because if the these dirs already exists with wrong group, it will only fix the group for the top level dirs, but not their contents.
1. We also don't want to do an unconditional `chown -R` on every single startup, since this is an expensive operation and might not even work depending on the fs.

As of this script, it will only create the top level dirs if they don't exist and it will only fix the group if the group is wrong.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (Mic92) [inline nixos/modules/services/web-apps/nextcloud.nix:94]: The only argument against removing this option is if your `nginx` is running somewhere else i.e. in a different host/container.

Comment (Mic92) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: The workaround would be `systemd.services.nginx.enable = false` in this case.

Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: You could still either setup a reverse proxy to the nginx inside the container, or disable it via `mkForce services.nginx.enable = false`. Do you think this is acceptable?

Comment (Mic92) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: I suppose though.

Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: But then you would run again into. https://github.com/NixOS/nixpkgs/issues/69261
Therefore it's not really done by switching that one option to false.
I think we need to decide. Either we want to support having nextcloud without nginx (then we need more config options) or we just don't.

Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: I would argue that the benefits we get by supporting non-nginx setups aren't big enough to justify the overhead it will have in maintaining it. It just makes things more complicated on every single change. When restricting to nginx in general we can just exclude a lot of question marks whenever we touch anything user/group related
 
Even if you run it inside a container, it shouldn't be a performance bottleneck to have this extra nginx running.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [reply nixos/modules/services/web-apps/nextcloud.nix:468]: That makes sense to me. Maybe consider `chgrp` if the intention is to only change the group.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (aanderse) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: Looking for @Ma27 to weigh in on this...

Comment (Mic92) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: I don't need it personally. We could also wait for someone to add it back ;). However it should get deprecated properly like here: https://github.com/NixOS/nixpkgs/blob/6eadc11005318fc4eef7da19898cff51689fb723/nixos/modules/tasks/network-interfaces.nix#L316

Comment (Ma27) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: Will think about this tonight @aanderse (oh an please ping me if I forget about this again :sweat_smile:  )

Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: Based on the current implementation, i think it would not be super hard to add support for non-nginx. We just need to add another config option to specify the user of the webserver used. This user will be added to the nextcloud group and phpfpm will be opened to that user.

I already had that implemented before but reverted it. I'm happy to re-revert ;).

The downside of it would be that it makes the module harder to understand for people and therefore harder to maintain i guess.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (Ma27) [inline nixos/modules/services/web-apps/nextcloud.nix:528]: What about `mkDefault true` here?

**Review by Ma27 — COMMENTED** [review]
The implementation seems fine: `nginx` is in most of the services in our tree that serve HTTP the default choice and now it's possible here to opt-out by disabling it.
Also did a brief test of the altered functionality and it appears fine to me.

However there are a few things I'd like to get resolved before merging this:

* We should mention this in our release-notes. When having a big library in your nextcloud, the `chown` will take some time, so admins should be aware of this *before* deploying the new config.

* Removed options must be documented with `mkRemovedOptionModule`.

* The nextcloud-specific tests don't evaluate because of the removed `nginx.enable` option.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (Ma27) [reply nixos/modules/services/web-apps/nextcloud.nix:94]: > The downside of it would be that it makes the module harder to understand for people and therefore harder to maintain i guess.

I think I mentioned this in a different thread already, but I might be wrong: I'm rather strongly against supporting multiple reverse-proxies for the exact reasons you gave, even more complexity (and thus more work for the maintainers - including me).

I think that your PR is actually a great solution: we have `nginx` as default (as most of the services here do), but admins now have the freedom to opt-out  of it and use an alternative.

**Comment by DavHau** [comment]
Things done:
 - deprecated `services.nextcloud.nginx.enable` option via mkRemovedOptionModule
 - replaced chown with chgrp
 - fixed tests by removing use of option `services.nextcloud.nginx.enable`
 - use mkDefault for `services.nginx.enable`

> We should mention this in our release-notes. When having a big library in your nextcloud, the chown will take some time, so admins should be aware of this before deploying the new config.

I don't think this will be necessary. Up until this point the setup service always did a chown an all nextcloud files on each startup. Therefore the new implementation doesn't bring any performance drawbacks compared to the old one.

**Review by aanderse — APPROVED** [review]
In general, LGTM :+1: I have not tested and don't use this module, though, so need someone like @Ma27 to approve as well.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (Ma27) [inline nixos/modules/services/web-apps/nextcloud.nix:51]: That's wrong: you can still use anything else (in fact now it's even easier than before!) by setting `services.nginx.enable = false`.

**Review by Ma27 — COMMENTED** [review]
LGTM :+1: 

* Did anyone test this with live data yet? I just checked whether the "migration" to the new config works fine in a temporary VM network.
* We should mention this in the release notes. The rollout of the new config can take quite long due to the recursive `chgrp`.

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (DavHau) [reply nixos/modules/services/web-apps/nextcloud.nix:51]: That's not entirely true. One also has to change the `listen.owner` and `listen.group` of phpfpm to something else than nginx and also add the user of the new webserver to `users.groups.nextcloud.members`. Should I add an example for disabling nginx to the deprecation text?

>We should mention this in the release notes. The rollout of the new config can take quite long due to the recursive chgrp

As I explained in my last comment, The rollout won't take any longer than the normal startup of the previous implementation. Therefore I think it's unnecessary to state anything in the release notes. Do you agree?

Comment (Ma27) [reply nixos/modules/services/web-apps/nextcloud.nix:51]: > Should I add an example for disabling nginx to the deprecation text?

Yes, please. But as this is possible (with a few overrides though), I'd also rephrase the message above.

> Therefore I think it's unnecessary to state anything in the release notes. Do you agree?

As mentioned above, I only tested this against a temporary environment. As I'm always slightly anxious when fiddling around with those kinds of scripts (touching user data) in NixOS, I'd prefer to test this against live-data first. I'll hopefully be able to do this tomorrow (with the fat library I'm hosting in my personal Nextcloud :P). After that I think that this should be good to go :)

Comment (Ma27) [reply nixos/modules/services/web-apps/nextcloud.nix:51]: Tested against my instance, didn't have any problems.

**Review by Ma27 — APPROVED** [review]

**Comment by ajs124** [comment]
This broke then manual: [log](https://nix-cache.s3.amazonaws.com/log/sm4ynvc1r7waxjaa8m3vpg450gvv5spi-nixos-manual-combined.drv). Plz fix, kthxbye. Or wait for a 10-12h, then I'll probably get around to it myself.

**Comment by aszlig** [comment]
@ajs124: Fixed in 1365b9ac700aabe8dc959e9897bc1376e416c553.

**Comment by DavHau** [comment]
Thanks @ajs124 .
@Ma27 I thought we still wanted to improve the phrase `The nextcloud module dropped support for other webservers than nginx.`.
But now the PR has already been merged. Should I make another commit ontop of this PR and repopen it?

**Comment by Ma27** [comment]
@aszlig @ajs124 sorry for that!
@DavHau yeah feel free to do that :) Otherwise I'll take care of this next week.

**Comment by Ma27** [comment]
Just for the record, this PR caused another regression in #95259 which got fixed in fddeb7cb73096786de8e59a8502cf3ae523bd694.

I'm fully aware that I merged this too early which is why I want to openly apologize to everyone who went into trouble because of the (now fixed) regressions I caused with the merge!

**Comment by GaetanLepage** [comment]
Hello,

I have read the manual on how to use a different webserver for nextcloud.
However, I don't really get why, to prevent nginx being configured for nextcloud, it is now necessary to **completely** disable nginx.
What if, I want to have the nginx service enabled for something else while still wanting to use, let's say httpd, as the web server for nextcloud ?

Thank you in advance for the clarification :)

**Pull request merged**
